tools/src/main/java/org/apache/kafka/trogdor/workload/RoundTripWorker.java

```diff
@@ -206,7 +208,9 @@ public void run() {
                     }
                     messagesSent++;
                     // we explicitly specify generator position based on message index
-                    ProducerRecord<byte[], byte[]> record = payloadGenerator.nextRecord(TOPIC_NAME, messageIndex);
+                    ProducerRecord<byte[], byte[]> record = new ProducerRecord(TOPIC_NAME, 0,
+                        KEY_GENERATOR.generate(messageIndex),
```

Comment:
Out of curiosity, why is the key generator not part of the spec also?

By the way, it would be nice to have some high-level javadocs for this class. I am not sure what a "round-trip" worker is supposed to do.

Reply:
Good question.  The reason why the key generator can't be configurable here is that RoundTripWorkload relies on the keys being sequential integers.  It uses this to ensure that it's gotten every message that was sent.  This wouldn't work with, say, ConstantGenerator (just to pick one example) since all the keys generated would be the same in that case.

We could potentially have a more sophisticated way of doing this that relied on a secondary interface.  So we could have `interface ReversableGenerator extends Generator`, and adds a function `long findPosition(byte[] value)`.  Then generators that allowed you to figure out the index from the payload (like sequential generator, and maybe some others) would implement ReversableGenerator, and ConstantGenerator, RandomGenerator, etc. would not.  This seems like overkill for now, though.

With regard to JavaDocs about RoundTripWorker: I agree.  I'll add it to the TODO